src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -220,6 +227,113 @@ export default class StudentApplication extends Component {
     this.setState({ updating: false });
   };
 
+  /**
+   * Callback for hall list remove button
+   * @param {String} hallSelectionValue The name of the hall that was selected
+   * @param {String|Number} hallRankValue The rank value that the user assigned to this hall
+   * @param {Number} index The index of the hall in the list
+   */
+  handleHallInputChange = (hallSelectionValue, hallRankValue, index) => {
+    console.log('Called "handleHallInputChange" in StudentApplication component'); //! DEBUG
+    console.log('hallName: ' + hallSelectionValue); //! DEBUG
+    console.log('hallRank: ' + hallRankValue); //! DEBUG
+    console.log('index: ' + index); //! DEBUG
+    this.setState({ updating: true });
+    if (index !== null && index >= 0) {
+      console.log('Attempting to update preferred halls'); //! DEBUG
+
+      let preferredHalls = this.state.preferredHalls; // make a separate copy of the array
+
+      // Create a new custom hallInfo object
+      let newHallInfo = preferredHalls[index];
+
+      // Error checking on the hallSelectionValue before modifying the newHallInfo object
+      if (
+        hallSelectionValue !== null &&
+        hallSelectionValue !== this.state.preferredHalls[index].hallName &&
+        this.state.preferredHalls.some((hallInfo) => hallInfo.hallName === hallSelectionValue)
+      ) {
+        // Display an error if the selected hall is already in the list
+        this.snackbarText = String(hallSelectionValue) + ' is already in the list.';
+        this.snackbarSeverity = 'info';
+        this.setState({ snackbarOpen: true });
+      } else if (hallSelectionValue !== null) {
+        // Create a new custom hallInfo object
+        newHallInfo.hallName = hallSelectionValue;
+      }
+
+      // Error checking on the hallRankValue before modifying the newHallInfo object
+      if (hallRankValue !== null && Number(hallRankValue) && Number(hallRankValue) >= 0) {
```

Comment:
This does work, since the return value of `Number(hallRankValue)` will be treated as false if it returns `0` or `NaN`, which is the intended behavior in this case.
As per Nick's suggestion, the Textfield for the rank value has since been replaced with a dropdown menu of integers, so we no longer need to verify whether the user entered a valid number for the rank value